resources/source-types/azure_blob.yaml

```
@@ -0,0 +1,91 @@
+apiVersion: bindplane.observiq.com/v1
+kind: SourceType
+metadata:
+  name: azure_blob
+  displayName: Azure Blob
+  icon: /icons/azure-blob-icon.svg
+  description: Receive logs and trace data from an Azure Blob Storage events.
+spec:
+  version: 0.0.1
+  supportedPlatforms:
+    - linux
+    - windows
+    - macos
+    - kubernetes-daemonset
+    - kubernetes-deployment
+    - openshift-daemonset
+    - openshift-deployment
+  parameters:
+    - name: connection_string
+      label: Connection String
+      description: The connection string to the Azure Storage account.
+      documentation:
+        - text: Read more
+          url: https://learn.microsoft.com/en-gb/azure/storage/common/storage-configure-connection-string
+      type: string
+      required: true
+
+    - name: event_hub_endpoint
+      label: Event Hub Endpoint
+      description: The Azure Event Hub endpoint triggering on the Blob Create events.
+      documentation:
+        - text: Read more
+          url: https://learn.microsoft.com/en-us/azure/storage/blobs/storage-blob-event-overview
+      type: string
+      required: true
+
+    - name: enable_logs
+      label: Enable Logs
+      description: Enable to collect logs.
+      type: bool
+      default: true
+
+    - name: logs_container
+      label: Container
+      description: Name of the Azure Storage container where logs are stored.
+      type: string
+      required: false
+      default: logs
+      relevantIf:
+        - name: enable_logs
+          operator: equals
+          value: true
+
+    - name: enable_traces
+      label: Enable Traces
+      description: Enable to collect traces.
+      type: bool
+      default: true
+
+    - name: traces_container
+      label: Container
+      description: Name of the Azure Storage container where traces are stored.
+      type: string
+      required: false
+      default: traces
+      relevantIf:
+        - name: enable_traces
+          operator: equals
+          value: true
+
+  logs:
```

I think this can be combined into one pipeline configuration 
```suggestion
  logs+traces:
```
The order might matter ^

---

It can be combined bit it also makes it so the toggles don't matter. I like if you toggle off traces it won't actually give you a trace exporter. If we combine them I think the best I can do is just not configure the trace container, which has a default value if omitted.

---

🤦 you are correct on that